hermes/probe/checknil/check_nil.go

```diff
@@ -0,0 +1,332 @@
+	}
+	return probe.NewProbeError(metrics.Success, nil)
+}
+
+// WriteNilFile writes the current in-memory journal proto a new nil file on
+// the target system. It then deletes the old nil file located on the system.
+// Arguments:
+//	- ctx: context so this probe can be cancelled if needed.
+//	- target: target run information.
+//	- client: initialised storage client for this target system.
+//	- logger: logger associated with the probe calling this function.
+// Returns:
+//	- err: error exit statuses are well defined in metrics.go
+// TODO(evanSpendlove): Add existing nil file deletion to here
+func UpdateNilFile(ctx context.Context, target *probe.Target, client stiface.Client, logger *logger.Logger) *probe.ProbeError {
+	file := &nilFile{}
+	var err error
+
+	file.contents, err = proto.Marshal(target.Journal)
+	if err != nil {
+		return probe.NewProbeError(metrics.ProbeFailed, err)
+	}
+
+	hash := sha1.New()
+	if _, err := io.Copy(hash, file.reader()); err != nil {
+		return probe.NewProbeError(metrics.ProbeFailed, fmt.Errorf("UpdateNilFile: unable to compute checksum of file contents: %w", err))
+	}
+
+	file.name = fmt.Sprintf("Hermes_Nil_%x", hash.Sum(nil))
+
+	bucket := target.Target.GetBucketName()
+
+	logger.Infof("UpdateNilFile(): Nil file object created, proceeding to upload.")
+
+	var writer stiface.Writer
+
+	if err := RecordAPILatency(target.LatencyMetrics.APICallLatency, metrics.APICreateFile, func() *probe.ProbeError {
+		writer = client.Bucket(bucket).Object(file.name).NewWriter(ctx)
+		_, err = io.Copy(writer, file.reader())
+		if err != nil {
+			switch err {
+			case storage.ErrBucketNotExist:
+				return probe.NewProbeError(metrics.BucketMissing, err)
+			default:
+				return probe.NewProbeError(metrics.ProbeFailed, err)
+			}
+			return probe.NewProbeError(metrics.APICallFailed, err)
+		}
+		return probe.NewProbeError(metrics.Success, nil)
+	}); err.Status != metrics.Success {
+		return err
+	}
+
+	if err := writer.Close(); err != nil {
+		return probe.NewProbeError(metrics.APICallFailed, err)
+	}
+
+	target.Journal.Intent = nil
+
+	logger.Infof("UpdateNilFile(): Nil file update complete.")
+
+	return probe.NewProbeError(metrics.Success, nil)
+}
+
+// RecordAPILatency records the time taken for the function passed to complete.
+// It then stores this in the metrics map passed using the provided labels.
+// Arguments:
+//	- m: api latency metric map for this target.
+//	- call: metric label for this API call
+//	- f: target function to be timed.
+// Returns:
+//	- err: error exit statuses are well defined in metrics.go
+// TODO(evanSpendlove): Migrate this into metrics.go and add a RecordProbeLatency() func there.
+func RecordAPILatency(m map[metrics.APICall]map[metrics.ExitStatus]*cpmetrics.EventMetrics, call metrics.APICall, f func() *probe.ProbeError) *probe.ProbeError {
+	start := time.Now()
+	err := f()
+	m[call][err.Status].Metric("hermes_api_latency_seconds").AddFloat64(time.Now().Sub(start).Seconds())
```

Comment:
"hermes_api_latency_seconds" should be a const, assuming it's used elsewhere in the code.

Reply:
I've refactored this into a constant.